File: JavaScript/leetcode75/src/test/Problem71_SearchSuggestionsSystem.test.js
const suggestedProducts = require("../main/Problem71_SearchSuggestionsSystem");

test("should return searched products", () => {
  const products = ["mobile", "mouse", "moneypot", "monitor", "mousepad"],
    searchWord = "mouse";

  const expected = [
    ["mobile", "moneypot", "monitor"],
    ["mobile", "moneypot", "monitor"],
    ["mouse", "mousepad"],
    ["mouse", "mousepad"],
    ["mouse", "mousepad"],
  ];

  expect(suggestedProducts(products, searchWord)).toStrictEqual(expected);
});

File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem71_SearchSuggestionsSystemTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem71_SearchSuggestionsSystemTest {

    private final Problem71_SearchSuggestionsSystem test = new Problem71_SearchSuggestionsSystem();

    @Test
    void suggestedProducts() {
        String[] products = new String[]{"mobile", "mouse", "moneypot", "monitor", "mousepad"};
        String searchWord = "mouse";

        List<List<String>> expect = Arrays.asList(List.of("mobile", "moneypot", "monitor"), List.of("mobile", "moneypot", "monitor"),
                List.of("mouse", "mousepad"), List.of("mouse", "mousepad"), List.of("mouse", "mousepad"));

        assertEquals(expect, test.suggestedProducts(products, searchWord));
    }
}